src/database/models/Product.ts

```diff
@@ -0,0 +1,105 @@
+import { Model, DataTypes } from 'sequelize';
+import sequelize from './index';
+import { UUIDV4 } from 'sequelize';
+import { Size } from './Size';
+import { Category } from './Category';
+
+export interface ProductAttributes {
+  id?: number;
+  sellerId: string;
+  name: string;
+  description: string;
+  images: string[];
+  color?: string[];
+  discount?: number;
+  price?: number;
+  quantity?: number;
+  expiryDate?: Date;
+  categoryId: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+export class Product extends Model<ProductAttributes> implements ProductAttributes {
+  public id!: number;
+  public sellerId!: string;
+  public name!: string;
+  public description!: string;
+  public discount!: number;
+  public quantity!: number;
+  public price!: number;
+  public expiryDate!: Date;
+  public categoryId!: string;
+  public images!: string[];
+  public color!: string[];
+  public readonly createdAt!: Date | undefined;
+  public readonly updatedAt!: Date | undefined;
+}
+
+Product.init(
+  {
+    id: {
+      type: DataTypes.UUID,
+      defaultValue: UUIDV4,
+      primaryKey: true,
+      unique: true,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    description: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    color: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: true,
+    },
+    discount: {
+      type: DataTypes.FLOAT,
+      allowNull: true,
+      defaultValue: 0,
+    },
+    price: {
+      type: DataTypes.FLOAT,
+      allowNull: true,
+    },
+    quantity: {
```

Comment:
this quantity also available in Size, why? why can't we have model which will combine those repeating variables in Products and Size?

Reply:
that's good idea though